test/test_charm.py

```diff
@@ -83,11 +46,12 @@ def __init__(self, *args: typing.Any):
             def _on_start(self, event: ops.EventBase):
                 self.started = True
 
+        framework = create_framework(monkeypatch, request)
```

Comment:
Since the event mocking is part of the function, it needs to appear before asserting events, hence moving it up here.